projectile.el

```diff
@@ -3577,32 +3577,25 @@ If the prefix argument SHOW_PROMPT is non nil, the command can be edited."
     (unless (string= command executed-command)
       (ring-insert command-history executed-command))))
 
-(defadvice compilation-find-file (around projectile-compilation-find-file)
+(defun compilation-find-file-projectile-find-compilation-buffer (marker filename directory &rest formats)
```

Comment:
the newly defined function should take the original definition as first argument; you should funcall this argument instead of compilation-find-file inside the body.

also, please keep the refactoring minimal (or split into commits) to ease review.

thanks!

Reply:
@mookid: right, will do. Thanks